File: aoj/volume0/n0013/Main.cpp
#include <iostream>
#include <stack>

using namespace std;

int main() {

    stack<int> stack;

    int num;
    while (cin >> num) {

        if (num == 0) {
            cout << stack.top() << endl;
            stack.pop();
        } else {
            stack.push(num);
        }

    }

    return 0;

}

File: aoj/volume0/n0013/Main.java
import java.util.Scanner;
import java.util.Stack;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<Integer>();

        while (sc.hasNextInt()) {

            int number = sc.nextInt();

            if (number == 0) {
                System.out.println(stack.pop());
            } else {
                stack.push(number);
            }

        }

    }

}
